### ISSUE TITLE
Testing CRD in the operator against validations in 1.18

### ISSUE THREAD

**Issue by brownleej** [FoundationDB/fdb-kubernetes-operator#300]
Kubernetes 1.18 has new validations that can cause issues with CRDs that embed the pod template type. We need to test our CRD against these validations and figure out what we need to do to get compatible with the latest Kubernetes versions.

Reference: https://github.com/kubernetes/kubernetes/issues/91395

**Comment by johscheuer** [comment]
The current generated version of the CRD's are fine:

```bash
$ kubectl apply -f ./config/crd/bases/
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
customresourcedefinition.apiextensions.k8s.io/foundationdbbackups.apps.foundationdb.org created
customresourcedefinition.apiextensions.k8s.io/foundationdbclusters.apps.foundationdb.org created
customresourcedefinition.apiextensions.k8s.io/foundationdbrestores.apps.foundationdb.org configured
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.1", GitCommit:"206bcadf021e76c27513500ca24182692aabd17e", GitTreeState:"clean", BuildDate:"2020-09-09T19:10:58Z", GoVersion:"go1.15.1", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.0", GitCommit:"e19964183377d0ec2052d1f1fa930c4d7575bd50", GitTreeState:"clean", BuildDate:"2020-08-26T14:23:04Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}
```

but when I run the following steps I get misconfigured CRDs (used `KubeBuilderVersion:"2.3.1"`):

```bash
# cleanup otherwise the manifests won't be newly generated
$ rm -rf ./config/crd/bases/*
$ make manifests
# now run the apply step again
```

I get the expected error from the issue:

```
for: "/Users/jscheuermann/go/src/github.com/FoundationDB/fdb-kubernetes-operator/config/crd/bases/apps.foundationdb.org_foundationdbclusters.yaml": CustomResourceDefinition.apiextensions.k8s.io "foundationdbclusters.apps.foundationdb.org" is invalid: [spec.validation.openAPIV3Schema.properties[spec].properties[initContainers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[processes].additionalProperties.properties[podTemplate].properties[spec].properties[initContainers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[processes].additionalProperties.properties[podTemplate].properties[spec].properties[containers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[podTemplate].properties[spec].properties[containers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[podTemplate].properties[spec].properties[initContainers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[containers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property]
```

Which should be solved by this PR: https://github.com/kubernetes-sigs/controller-tools/pull/480 with the new version of `controller-gen` (devel containing the patch) everything works fine and the operator is able to create clusters.

TLDR: once a new `controller-gen` version is out and `kubebuilder` uses it the CRDs must be newly created. I didn't test the migration path from the CRD `v1beta1` to `v1` but this will probably an interesting challenge to tackle.
